### ISSUE TITLE
update flags in to be more consistent between a workflow and plantcv-workflow.py

### ISSUE THREAD

**Issue by dschneiderch** [danforthcenter/plantcv#470]
New issue for discussion about flags in workflows vs plantcv-workflow.py in parallel package from pull request #457 

I think all the flags are quite confusing. In particular, its confusing that the same letter can mean different things in my workflow vs plantcv-workflow. Anything to standardize them and make them more intuitive would make it easier for people to use.

in VIS workflow:

-i, --image = Input image file
-o, --outdir = output directory
-r, --result = result file
-w, --writeimg = write out image (although you could use this as a T/F flag for anything)
-D, --debug = debug state

in parallel:

-d, --dir = directory of input images
-p, --workflow = workflow to use
-i, --outdir = output directory
-D, --dates = dates to select

so in particular, the d s, i s, and o s
I think it would be more intuitive if they were universal where possible:

-i,--input (this is an image in vis-workflow, a directory for parallel workflow)
-d, --dates = dates to select
-o, --outdir = output directory
-D, --debug = debug state (workflows only)
-O, --debugout = debug_outdir (workflows only)

I will also note that sample_images uses another set of i,o flags:

-s, --source = source or input directory
-o, --outdir = output directory

I would replace -s with -i

finally, yes, I think these would be good (although I was going to ask about the above how you felt about having them be case sensitive.) Personally I like to use the flagnames so I don' t have to remember what each letter means. The benefit of -p, --processes is if you ever want to add -T, --threads then its available. processes is recognizable IMHO.

Workflow script file: -w, --workflow 
Parallel processes: -p, --processes
Save analysis images: -W, --writeimg

last, i find -c, --create a little awkward. at first I thought that it wouldn't make the results file unless i had -c. This is really --overwrite if I understand correctly. -f for force since we already use -o? - but then --meta might have to become -m

I would also like to specify -L (or similar) to specify a parent directory for the timestamped directories that are created with every run of `plantcv-workflow.py`

**Comment by nfahlgren** [comment]
I'm writing up a couple related issues and have a question that we have sort of discussed before. Should we keep developing/refining the input flags to `plantcv-workflow.py` or would it be easier for users (especially on different platforms) if we instead adopted a configuration file-based system for configuring workflow execution? It would be the same kind of inputs but instead of using argparse with the short and long flags we could just have a simple key-value pair file:

```
input       =
workflow    =
metadata    = 
dates       =
outdir      =
debugout    =
processes   = 
imgformat   =
delimiter   =   
match       =
coprocess   =
writeimg    =
other_args  =
```

Something like that. Then when stuff like a regex is input into the file we don't have to worry about how characters are treated on various command-line systems. Plus like the shell and bat scripts, you get a clean recorded record of what you ran.

Downside is maybe that it's a little bit harder to get help since you can't just use `--help`. But we could have `plantcv-workflow.py --init_config <config file>` or something to create an empty template file.

**Comment by dschneiderch** [comment]
+1 for config file. I only ever run plantcv-workflow.py from run_workflow.cmd that has all my flags in it anyway.

presumably we can still have --help flag for plantcv-workflow.py that describes the config file.

**Comment by nfahlgren** [comment]
Cool, my intuition is that people will broadly feel more comfortable with this than huge commands with many flags. But anyone should definitely chime in if they disagree.
